### ISSUE TITLE
Support for SUSE?

### ISSUE THREAD

**Issue by juniorsysadmin** [treydock/puppet-gpg_key#5]
Currently rpm.rb has:

> confine :osfamily => :redhat

Are there any plans to support non-RedHat systems which support RPM, like SUSE?

**Comment by treydock** [comment]
@juniorsysadmin Supporting more than RedHat systems is possible.  Do you have a public repo for SUSE that has a GPG key I can add to the acceptance tests?  I see Puppetlabs has SLES 11p1 vagrant box available which I can add to the acceptance tests.  I've never directly supported SUSE simply because I have no means to test against that OS but if I can get an acceptance test added then I'd extend the support to those operating systems.

**Comment by juniorsysadmin** [comment]
The SUSE SolidDriver Program http://drivers.suse.com/doc/Usage/Package_Signing_Key.html
seems to be one.

**Comment by juniorsysadmin** [comment]
Wait, do we care about where the key comes from? The format of a public key should be the same.

**Comment by treydock** [comment]
@juniorsysadmin Where it comes from is not important as long as it's public and I won't get in trouble for having the file as a fixture in this module.  I'll see about updating the acceptance tests to download the keys rather than committing them as part of this repo.

**Comment by alvagante** [comment]
+ 1 for Suse support. FYI, tests aside, the diff below is enough to make the module work on Suse:

```
diff --git a/lib/puppet/provider/gpg_key/rpm.rb b/lib/puppet/provider/gpg_key/rpm.rb
index 4cca540..93403a4 100644
--- a/lib/puppet/provider/gpg_key/rpm.rb
+++ b/lib/puppet/provider/gpg_key/rpm.rb
@@ -11,7 +11,7 @@ Puppet::Type.type(:gpg_key).provide(:rpm) do
   end

   defaultfor :osfamily => :redhat
-  confine :osfamily => :redhat
+  confine :osfamily => [ :redhat, :suse ]
```

**Comment by treydock** [comment]
I've released 0.1.0 with this change.
